temba/utils/analytics/prometheus.py

```diff
@@ -0,0 +1,53 @@
+import logging
+
+import zope.interface
+from prometheus_client import Counter, Gauge
+
+from django.conf import settings
+
+from temba.utils.analytics.base import IMetricBackend
+
+logger = logging.getLogger(__name__)
+
+
+MESSAGE_COUNTS = Gauge("message_counts", "Current count of messages", ["hostname", "direction", "state"])
+RELAYER_SYNC = Gauge("relayer_sync_seconds_duration", "Duration of relayer sync view", ["hostname"])
+CONTACTS = Counter("contacts_total", "Number of contacts", ["hostname", "action"])
+
+
+@zope.interface.implementer(IMetricBackend)
+class PrometheusBackend:
+    """
+    A metrics backend for Prometheus
+    """
+
+    def _get_hostname(self):
+        # settings.HOSTNAME is actually service name (like textit.in), and settings.MACHINE_NAME is the name of the machine
+        # (virtual/physical) that is part of the service
+        return "%s.%s" % (settings.MACHINE_HOSTNAME, settings.HOSTNAME)
+
+    def gauge(self, event, value=None):
+        if value is None:
+            value = 1
+
+        if event.startswith("temba.current_outgoing"):
+            MESSAGE_COUNTS.labels(hostname=self._get_hostname(), direction="outbound", state=event.split("_")[-1]).set(
+                value
+            )
+        elif event.startswith("temba.current_incoming"):
+            MESSAGE_COUNTS.labels(hostname=self._get_hostname(), direction="inbound", state=event.split("_")[1]).set(
+                value
+            )
+        elif event == "temba.relayer_sync":
+            RELAYER_SYNC.labels(hostname=self._get_hostname()).set(value)
+        else:
+            logger.warning(f"Unknown prometheus gauge metric {{event}} with value {{value}}")
+
+    def increment(self, event, value=None):
+        if value is None:
+            value = 1
+
+        if event == "temba.contact_created":
+            CONTACTS.labels(hostname=self._get_hostname(), action="created").inc(value)
```

Comment:
`created` label seems a bit redundant here, though maybe that's me no understanding prometheus properly.

Reply:
This is more for if we want to at a later stage add updated/deleted actions, without having to change dashboard setups. I guess you can argue either way, right now it's not needed, the examples usually prefer having a single metric for the object, and having a label for each CRUD method.

Happy to do either method, since it doesn't matter at this moment with just a single action type.

Reply:
Oh actually I missed that this was just recording a contacts counter, I thought it was using `contact_created` still. No, this seems great then actually.

One thing though is maybe these should still be prefixed with `temba_`? Grafana seems to build tree-ish browsing based on the underscore so it may make it a bit easier to look for rapidpro related events.